draft-ietf-ppm-dap.md

```diff
@@ -1837,11 +1861,18 @@ struct {
 } PrepareResp;
 ~~~
 
-Next, the Helper constructs an `AggregationJobResp` message
-({{aggregation-helper-init}}) with each prep step. The order of the prep steps
-MUST match the Leader's request. It responds to the Leader with HTTP status 200
-OK, media type `application/dap-aggregation-job-resp`, and a body consisting of
-the `AggregationJobResp`.
+After receiving the response to its `AggregationJobContinueReq`, the Leader
+makes an HTTP GET request to the aggregation job URI to check on the status
+of the aggregation job. If the continuation is not finished yet, the Helper
+responds with HTTP status 202 Accepted. The response MAY include a Retry-After
+header field to suggest a polling interval  to the Leader.
+
+If the aggregation job is ready, the Helper constructs an `AggregationJobResp`
+message ({{aggregation-helper-init}}) with each prep step. The order of the
+prep steps MUST match the Leader's `AggregationJobContinueReq`. It responds
```

Comment:
Interesting implication of this is that it means the helper must keep track of the order in which prep steps occurred in the request. Janus already does this, but perhaps there are implementations that relied upon having the leader request in memory.

Reply:
A point in favor of https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/217, but I doubt that discussion needs to be rehashed.

Reply:
I don't think this requirement is any harder to meet as a result of taking this PR. Or am I missing something?

Reply:
Hmm. In the Leader Continuation section, we have:

> The PrepareContinues MUST be in the same order as the previous aggregate request.

But in the Helper Continuation section, we have:

> Additionally, if any prep step appears out of order relative to the previous request, then the Helper MAY abort with error invalidMessage.

That is, we proscribe that the Leader MUST respect the original order of reports, but only that the Helper MAY respect the original order (and therefore MAY not). I'm not sure if there is a good reason for this mismatch -- it seems to me that this is Postel's Law ("be conservative about what you send, liberal in what you accept") thinking which is recognized as [causing reliability & security issues](https://en.wikipedia.org/wiki/Robustness_principle#Criticism). IMO, the Helper should require that reports be in the same order if the Leader does, i.e. the quoted MAY should be a MUST instead.

(That said, I do think the ordering requirement is perhaps ill-justified, but it's not important enough to rehash #217.)

Reply:
I agree it's an oversight. It should be MUST.